### PR TITLE
NEWS: some improvements and additions

### PR DESCRIPTION
**Pull request by StefanKarpinski**

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -68,7 +60,8 @@ Language changes
   * `using` and `import` are now case-sensitive even on case-insensitive filesystems
     (common on Mac and Windows) ([#13542]).
 
-  * Relational symbols are now allowed as infix operators ([#8036]).
+  * Relational algebra symbols are now allowed as infix operators ([#8036]):
+    `⨝`, `⟕`, `⟖`, `⟗` a for joins and `▷` for anti-join.
```
Comment (tkelman) [inline NEWS.md]: "a" extraneous in "a for"?

File: NEWS.md [hunk 2]
```diff
@@ -77,7 +70,8 @@ Language changes
     This also applies to the `>:` operator.
 
   * Simple 2-argument comparisons like `A < B` are parsed as calls instead of using the
-    `:comparison` expression type ([#15524]).
+    `:comparison` expression type ([#15524]). The `:comparison` expresison type is still
```
Comment (tkelman) [inline NEWS.md]: typo, expresison -> expression

File: NEWS.md [hunk 3]
```diff
@@ -77,7 +70,8 @@ Language changes
...
+    produced in AST when comparisons are chained (e.g. `A < B ≤ C`).
```
Comment (tkelman) [inline NEWS.md]: in ASTs

File: NEWS.md [hunk 4]
```diff
@@ -127,9 +130,10 @@ Breaking changes
     If a reshaped copy is needed, use `copy(reshape(a))` or `copy!` to a new array of
     the desired shape ([#4211]).
 
-  * `mapslices` now re-uses temporary storage. Recipient functions
-    that expect input slices to be persistent should copy data to
-    other storage ([#17266]).
+  * `mapslices` now re-uses temporary storage. Recipient functions that expect
```
Comment (tkelman) [inline NEWS.md]: did this get reverted?

File: NEWS.md [hunk 5]
```diff
@@ -193,15 +208,16 @@ Library improvements
   * Arrays and linear algebra:
 
     * All dimensions indexed by scalars are now dropped, whereas previously only
-      trailing scalar dimensions would be omitted from the result ([#13612]).
+      trailing scalar dimensions would be omitted from the result ([#13612]). This
+      is a very major behavioral change, but should cause obvious failures.
```
Comment (tkelman) [inline NEWS.md]: mention the `i:i` trick to get the old behavior?

File: NEWS.md (hunk 4 above)
Comment (timholy) [inline NEWS.md]: Still accurate unless there's something newer than #17266. What got changed is that the new one makes it impossible to overwrite the input array.

File: NEWS.md [hunk 6]
```diff
@@ -105,19 +99,28 @@ Compiler/Runtime improvements
...
+
 Breaking changes
 ----------------
 
   * The assignment operations `.+=`, `.*=` and so on now generate calls
     to `broadcast!` on the left-hand side (or call to `view(a, ...)` on the left-hand side
-    if the latter is a `a[...]` expression.  This means that they will fail
-    if the left-hand side is immutable (or does not support `view`), and will
-    otherwise change the left-hand side in-place ([#17510], [#17546]).
+    if the latter is an indexing expresion, (e.g. `a[...]`). This means that they will fail
```
Comment (ararslan) [inline NEWS.md]: "expresion" should be "expression"

**Comment by tkelman** [comment]
NEWS also needed for https://github.com/JuliaLang/julia/pull/16024

File: NEWS.md [hunk 7]
```diff
@@ -148,7 +152,18 @@ Library improvements
...
-      a string contains only ASCII characters.
+      a string contains only ASCII characters. The `ascii(s)` function now
+      converts `s` to `String`, raising an `ArgumentError` exception if `s` is
+      not pure ASCII.
+
+    * The `UTF16String` and `UTF32String` types and corresponding `utf16` and
+      `utf32` converter functions have been removed from the standard library.
+      If you need these types, they have been moved to the `LegacyStrings`
+      package. In the future, more robust Unicode string support will be provided
+      by the `StringEncodings` package. If you only need these types to call wide
```
Comment (omus) [inline NEWS.md]: Would be nice if the referenced packages were linked to.

File: NEWS.md [hunk 8]
```diff
@@ -157,6 +175,14 @@ Library improvements
     * A `transcode(T, src)` function is now exported for converting data
       between UTF-xx Unicode encodings ([#17323]).
 
+    * Comparisons between `Char`s and `Integer`s are now deprecated ([#16024]):
+      `'x' == 120` now produces a warning but still evaluates to `true`. In the
+      future it may evalaute to `false` or the comparison may be an error. To
```
Comment (tkelman) [inline NEWS.md]: evaluate

File: NEWS.md [hunk 9]
```diff
@@ -157,6 +175,14 @@ Library improvements
...
+      compare characters with integers you should either convert the integer to
+      a character value or conver the character the corresponding code point
```
Comment (tkelman) [inline NEWS.md]: convert

**Pull request merged**